src/builtin.jq

```diff
@@ -12,24 +12,24 @@ def add: reduce .[] as $x (null; . + $x);
 def del(f): delpaths([path(f)]);
 def _assign(paths; $value): reduce path(paths) as $p (.; setpath($p; $value));
 def _modify(paths; update):
-    reduce path(paths) as $p (.;
+    reduce path(paths) as $p ([., []];
         . as $dot
       | null
       | label $out
-      | ($dot | getpath($p)) as $v
+      | ($dot[0] | getpath($p)) as $v
       | (
           (   $$$$v
             | update
             | (., break $out) as $v
             | $$$$dot
-            | setpath($p; $v)
+            | setpath([0] + $p; $v)
           ),
           (
               $$$$dot
-            | delpaths([$p])
+            | setpath([1]; .[1] + [$p])
```

Comment:
Lots of `memcpy()`s here.  How about you append to the array of paths to be deleted then...

Reply:
> How about you append to the array of paths to be deleted then...

How can I achieve that? `.[1] += [$p]` is after compiled to `_modify` so using `setpath` is better IMO. Or can I avoid array boxing for `[$p]` somehow?

Reply:
You can just `setpath([length]; $p)`.

Reply:
Done: bb9abca35dcd0028bfb3ce7493c065d904ce5aaa.